staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -498,8 +503,69 @@ func GetFieldManagerFlag(cmd *cobra.Command) string {
 	return GetFlagString(cmd, "field-manager")
 }
 
+type DryRunStrategy int
+
+const (
+	DryRunNone DryRunStrategy = iota
+	DryRunClient
+	DryRunServer
+)
+
+// TODO(jmodes): remove
 func GetDryRunFlag(cmd *cobra.Command) bool {
-	return GetFlagBool(cmd, "dry-run")
+	dryRunStrategy, err := GetDryRunStrategy(cmd)
+	if err != nil {
+		klog.Fatalf("error accessing --dry-run flag for command %s: %v", cmd.Name(), err)
+	}
+	if dryRunStrategy == DryRunServer {
+		klog.Fatalf("--dry-run=server for command %s is not supported yet", cmd.Name())
```

Comment:
OK That's great, but we want to get rid of this before the release I think!

Reply:
Agreed!